Template/function_definitions.inc

```diff
@@ -1,44 +1,41 @@
-//==========================================================================
-// Class member functions for calculating the matrix elements for
-%(process_lines)s
-
-//--------------------------------------------------------------------------
 // Initialize process. 
   
-%(process_class)s::%(process_class)s(Parameters_%(model_name)s &params):
-    params(params)
-    {
-    %(initProc_lines)s
+%(process_class)s::%(process_class)s(const ParameterSet& configuration) {
+    
+    std::string param_card = configuration.get<std::string>("card");
+    params.reset(new Parameters_%(model_name)s(SLHA::Reader(param_card)));
+  
+    %(constructor_lines)s
 
     %(finalstates_map)s
 } 
 
 //--------------------------------------------------------------------------
 // Evaluate |M|^2, return a map of final states 
 
-std::map< std::pair<int, int>, double > %(process_class)s::sigmaKin( const std::vector< std::vector<double> > &initialMomenta, const std::vector< std::pair<int, std::vector<double> > > &finalState ) { 
+std::map< std::pair<int, int>, double > %(process_class)s::compute( const std::pair< std::vector<double>, std::vector<double> > &initialMomenta, const std::vector< std::pair<int, std::vector<double> > > &finalState ) { 
 
     // Set initial particle momenta
-	momenta[0] = (double*) (&initialMomenta[0][0]);
-    momenta[1] = (double*) (&initialMomenta[1][0]);
+	momenta[0] = (double*) (&initialMomenta.first[0]);
+    momenta[1] = (double*) (&initialMomenta.second[0]);
 
     // Suppose final particles are passed in the "correct" order
     std::vector<int> selectedFinalState(%(nexternal)s - 2);
     size_t index = 2;
-    for (auto const &finalPart: finalState){
+    for (auto const &finalPart: finalState) {
```

Comment:
In order to fix https://github.com/MoMEMta/MoMEMta-MaGMEE/pull/3#issuecomment-219662965, I'd propose to change the `foreach` loop to an old style loop. We know here how many final particles we expect (`%(nexternal)s - 2` or `selectedFinalState.size()`), so we can use:

```
for (size_t index = 0; index < (%(nexternal)s - 2); index++) {
    selectedFinalState[index] = finalPart.first;
    momenta[index + 2] = (double*) (&finalPart.second[0]);
}
```

This at least would solve the *too much particles*